ironfish/src/account/accounts.ts

```diff
@@ -616,12 +616,18 @@ export class Accounts {
   ): Promise<ReadonlyArray<{ hash: string; note: Note; index: number | null }>> {
     const unspentNotes = []
 
+    const notePromises = []
     for (const transactionMapValue of this.transactionMap.values()) {
-      const result = await this.workerPool.getUnspentNotes(
-        transactionMapValue.transaction.serialize(),
-        [account.incomingViewKey],
+      notePromises.push(
+        this.workerPool.getUnspentNotes(transactionMapValue.transaction.serialize(), [
+          account.incomingViewKey,
+        ]),
       )
+    }
```

Comment:
Any reason this for loop couldn't be a `.map`?

```
const notePromises = this.transactionMap.values.map(
  x => this.workerPool.getUnspentNotes(
    x.transaction.serialize(), [account.incomingViewKey]
  )
)
```

Reply:
my js-fu is disgraceful 😢

Reply:
Oh, jk this is a generator, not an iterator